image/postgres/scripts/init-entrypoint.sh

```diff
@@ -5,4 +5,50 @@ set -Eeo pipefail
 # Initialize DB if it does not exist
 if [ ! -s "$PGDATA/PG_VERSION" ]; then
   initdb --auth-host=scram-sha-256 --auth-local=scram-sha-256 --pwfile /run/secrets/stackrox.io/secrets/password --data-checksums
+else
+    # Verify if we need to perform major version upgrade
+    PG_BINARY_VERSION=$(postgres -V |\
+        sed 's/postgres (PostgreSQL) \([0-9]*\).\([0-9]*\).*/\1/')
+
+    if [ $(cat "${PGDATA}/PG_VERSION") -lt "$PG_BINARY_VERSION" ]; then
+        # Data version is less than binaries, upgrade
+
+        # Do a backup first, ideally move it to a separate volume. Since the
+        # database is stopped we may as well simple take a filesystem backup.
+        # Alternatives would be pg_dump or pg_basebackup, both require running
+        # database cluster.
+        echo "Backup..."
+        export BACKUP_DIR="/tmp/backups/$(date +%s)"
+        mkdir -p $BACKUP_DIR
+        tar -cf $BACKUP_DIR/backup.tar -C $PGDATA .
```

Comment:
If the init container fails for whatever reason, kubelet will restart it, and it will quickly fill up the filesystem with backup tarfiles. I guess the code should create a marker file once `tar` exits successfully, and skip the command on subsequent executions...?

Reply:
Yeah, that's a good point.

Reply:
The backup path now contains postgres versions, and if exists no more backups are taken.